app/assets/javascripts/discourse/app/routes/users.js

```diff
@@ -35,12 +36,20 @@ export default DiscourseRoute.extend({
     }
   },
 
+  afterModel(model) {
+    return Group.findAll().then((groups) => {
```

Comment:
I don't think this is right because it returns all visible groups but we need to check for groups where the members are visible as well. If the members are not visible to the user, we should not show the group.

Reply:
Indeed a nice catch. Added a filter of `can_see_members` to this list.